File: szavazas-core/src/main/java/hu/kdea/szavazas/ballotprocessor/common/Inverter.java
package hu.kdea.szavazas.ballotprocessor.common;

import boofcv.struct.image.GrayU8;

public final class Inverter {
    private Inverter() {
    }

    public static GrayU8 invert(GrayU8 gray) {
        GrayU8 out = new GrayU8(gray.width, gray.height);
        for (int y = 0; y < gray.height; y++) {
            for (int x = 0; x < gray.width; x++) {
                out.set(x, y, 255 - gray.get(x, y));
            }
        }
        return out;
    }
}

File: szavazas-core/src/main/java/hu/kdea/szavazas/ballotprocessor/common/Inverter.kt
package hu.kdea.szavazas.ballotprocessor.common

import boofcv.struct.image.GrayU8

object Inverter {
    fun invert(gray: GrayU8): GrayU8 {
        val out = GrayU8(gray.width, gray.height)
        for (y in 0 until gray.height)
            for (x in 0 until gray.width)
                out.set(x, y, 255 - gray.get(x, y))
        return out
    }
}